cirq-core/cirq/qis/states.py

```diff
@@ -115,38 +166,14 @@ def state_vector(self) -> Optional[np.ndarray]:
             return None
         return np.reshape(self.data, (self._dim,))
 
-    def state_tensor(self) -> Optional[np.ndarray]:
-        """Return the state tensor of this state.
-
-        A state tensor stores the amplitudes of a pure state as an array with
-        shape equal to the qid shape of the state.
-        If the state is a density matrix, this method returns None.
-        """
-        if self._is_density_matrix():
-            return None
-        return np.reshape(self.data, self.qid_shape)
-
     def density_matrix(self) -> np.ndarray:
```

Comment:
Keeping this here seems wrong - density matrices should override it, and everything else can call the parent method.